readthedocs/rtd_tests/tests/test_post_commit_hooks.py

```diff
@@ -8,6 +8,101 @@
 log = logging.getLogger(__name__)
 
 
+class GitLabWebHookTest(TestCase):
+    fixtures = ["eric", "test_data"]
+
+    def tearDown(self):
+        tasks.update_docs = self.old_bd
+
+    def setUp(self):
+        self.old_bd = tasks.update_docs
+
+        def mock(*args, **kwargs):
+            log.info("Mocking for great profit and speed.")
+        tasks.update_docs = mock
+        tasks.update_docs.delay = mock
+
+        self.client.login(username='eric', password='test')
+        self.payload = {
+            "object_kind": "push",
+            "before": "95790bf891e76fee5e1747ab589903a6a1f80f22",
+            "after": "da1560886d4f094c3e6c9ef40349f7d38b5d27d7",
+            "ref": "refs/heads/awesome",
+            "user_id": 4,
+            "user_name": "John Smith",
+            "user_email": "john@example.com",
+            "project_id": 15,
+            "repository": {
+                "name": "Diaspora",
+                "url": "git@github.com:rtfd/readthedocs.org.git",
```

Comment:
@gregmuellegger, this payload data is based on the [POST example](https://gitlab.com/gitlab-org/gitlab-ce/blob/master/doc/web_hooks/web_hooks.md) provided by GitLab.

The modifications introducing `github` were necessary to pass the tests, as the project lookup matches the git url against `github.com/rtfd/readthedocs.org`.